dulwich/porcelain.py

```diff
@@ -2177,11 +2177,11 @@ def ls_files(repo):
         return sorted(r.open_index())
 
 
-def find_unique_abbrev(object_store, object_id):
+def find_unique_abbrev(object_store: ObjectContainer, object_id: str, abbrev: int | None = 7) -> str:
```

Comment:
find_unique_abbrev is meant to find the unique abbreviation for a commit; passing in the length is counter to that. Perhaps just use ``[:abbrev]`` if abbrev is specified and fall back to find_unique_abbrev otherwise?

Reply:
It should be in find_unique_abbrev, but it should be return a commit that's as long as necessary to prevent ambiguity - not something that the caller determines.

Reply:
> It should be in find_unique_abbrev, but it should be return a commit that's as long as necessary to prevent ambiguity - not something that the caller determines.

Ok; to check if I interpret that correctly: a future complete implementation of `find_unique_abbrev` would handle ambiguous-commit avoidance, but `describe` can simply return the caller-specified `abbrev` prefix length.

Reply:
Lgtm, can you mark the pr as no longer a draft?

Reply:
Done; thanks!